### DISCUSSION TITLE
Odd behaviour when switching sources

### DISCUSSION THREAD

**Discussion by sjl68** [awawa-dev/HyperHDR#743] [General]
I have Libreelec on mini pc (Intel I3,Nvidia GT1030 GPU. the HyperHDR addon is installed with the suspend mod (thanks again).

The capture stick is a MS2130 and this all then feeds into a Feintech HDMI Matrix Switch, attached to that is a Nintendo Switch, a Nintendo Wii U and a Steamdeck.

When all connected if I switch between sources the LED's are in blue during the switching, additionally I have to swith back manually to the previous device i.e LE is on and I turn the Switch on during the transition period while the Switch is booting the LED's then go a pale dim blue. Put the Switch to sleep and the HDMI Matrix does not go back to the LE pc.
That behaviour occurs on every device. Tonight I tested a few things and found that if I disable HyperHDR then the switching back to LE works as expected and the switching from LE to a new source does not illuminate the LED's in the pale blue colour.

Is this a setting issue? Or is it default behaviour for HyperHDR?

Thanks :)

**Comment by awawa-dev** [comment]
Hi

Is this what the connection looks like?  Nintendo Switch | Nintendo Wii U | Steamdeck | PC Libreelec => Feintech HDMI Matrix Switch
=> MS2130 | TV

> Put the Switch to sleep and the HDMI Matrix does not go back to the LE pc.

If I understood correctly, in the case of this particular behavior, the problem is with the Feintech matrix and because it only manages the image that is received by the MS2130. When switching sources, the splitter may generate some temporary boards in a certain color for a while (few seconds max) and it can cause some strange reaction from LEDs, but you shouldn't worry about that.

MS2130 has one problem that can probably be solved with a firmware (during testing), but so far it has manifested itself with Ezcoo splitters: when switching sources or turning on/off the grabber or splitter, when reactivating capturing, the image was much darker than before. However, I don't know if that is the case here. You need to watch the 'live preview' in HyperHDR to see what causes the change in the LEDs.

Thank you again for your support :)

**Reply by sjl68** [reply]
That makes sense, I just tried live preview and when it switches to another source the video stays black which I guess indicates its either the matrix or the MS2130.
I think I have every firmware for the Feintech so will test some of them out just to see if it makes any difference - its not a deal breaker :)

My pleasure 💯

**Reply by sjl68** [reply]
@awawa-dev just to follow this up as I have been messing around with different options and just trying things out.

If I install the LUT table for MS2130 YUV as per the pop up message then I get the blue LED behaviour when switching between sources. If I delete or rename the new LUT file then its uses the default one and I do **not** get the blue LED's come on when switching between sources.

Additionally I worked out that the reason for the matrix not auto switching is due to the SteamDeck - I always choose sleep when I shut it down. I found that by unplugging the HDMI cable from either the Steam Deck dock or from the matrix it allows the auto switching to work. 
My quick work around is to add a zigbee plug to the Deck wall plug and then I added a ping sensor in Home Assistant to monitor the activity and it now disconnects the power to the Deck dock allowing the auto switch to work. It also works in reverse when starting the Deck.

I am waiting for a Zigbee IR blaster to arrive from Ali Express as a neater solution (hopefully) that will switch the matrix back to the main source (Libreelec)

**Reply by awawa-dev** [reply]
Maybe your source is using YUV/RGB limited 16-235 instead of the standard full range 0-255, so no black level for such setup only some strange dark colors instead. Try to use different LUT MS2130 for YUV limited that is available to download or switch you video source to full range.

**Reply by sjl68** [reply]
Well according to Libreelec I am using the standard full range. Using the LUT limited does work as expected -thank you again.

**Reply by awawa-dev** [reply]
It's possible that the splitter messed something up in the signal output to the USB grabber (it may also depend on the version of the ms2130 firmware and what protocol is negotiated with the splitter) because it may be different from signal format the TV will receive. Although it is better to make sure that it has the full range in the TV diagnostic menu and the splitter does not limit the image quality on the TV.
